### PR TITLE
Create WorkspaceTagsSettingsPage

### PR DESCRIPTION
**Pull request by waterim**
### Details

Create Tags Settings Page

### Fixed Issues

$ https://github.com/Expensify/App/issues/37310
$ https://github.com/Expensify/App/issues/37848
$ https://github.com/Expensify/App/issues/37873

### Tests

1. Go to settings in collect policy workspace
2. Go to tags
3. Click on Settings button
4. Try to update Tags name
5. Try the switcher

-----

1. Go to tag page
2. try to reload the page
3. Look that sidebar is correct.

------

1. Navigate to staging.new.expensify.com
2. Go to Workspace settings > Collect workspace
3. Select the "Tags" tab
4. Add some tags via classic app
5. Refresh the page to see the newly added tags

1. 
- [x] Verify that no errors appear in the JS console

### Offline tests

Same as tests

### QA Steps

None

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [x] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos

<details>
<summary>Web</summary>
image
image

</details>
<details>
<summary>IOS</summary>
image
image
</details>

<details>
<summary>Android</summary>
image
image
</details>

### CODE REVIEW

**Comment by luacmartins** [comment]
Assigning @mkhutornyi for review!

**Review by mkhutornyi — CHANGES_REQUESTED** [review]

**Comment by waterim** [comment]
@mkhutornyi Updated and resolved

**Comment by waterim** [comment]
@mkhutornyi Updated everything, also fixed fonts and bugs

**Comment by waterim** [comment]
@mkhutornyi updated

File: src/libs/actions/Policy.ts [hunk 1]
```diff
@@ -1846,6 +1847,19 @@ function openPolicyCategoriesPage(policyID: string) {
     API.read(READ_COMMANDS.OPEN_POLICY_CATEGORIES_PAGE, params);
 }
 
+function OpenPolicyTagsPage(policyID: string) {
```
Comment (luacmartins) [inline src/libs/actions/Policy.ts:1850]:
```suggestion
function openPolicyTagsPage(policyID: string) {
```

File: src/libs/actions/Policy.ts [hunk 2]
```diff
@@ -2539,6 +2553,99 @@ function setWorkspaceRequiresCategory(policyID: string, requiresCategory: boolea
...
 
+function setPolicyRequiresTag(policyID: string, requiresTag: boolean) {
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                value: {
+                    requiresTag,
+                    errors: {
```
Comment (luacmartins) [inline src/libs/actions/Policy.ts:2564]: I think this should be `errorFields`. `errors` is just an object keyed by timestamp, so this wouldn't work as we have here. Same for the others below.

File: src/libs/actions/Policy.ts [hunk 3]
```diff
@@ -2539,6 +2553,99 @@ function setWorkspaceRequiresCategory(policyID: string, requiresCategory: boolea
...
+    const oldName = policyTagListName.oldName;
+    const oldPolicyTags = policyTags?.[oldName] ?? {};
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
+                value: {
+                    [newName]: {...oldPolicyTags, name: newName},
+                    [oldName]: null,
```
Comment (luacmartins) [inline src/libs/actions/Policy.ts:2621]: Hmm I think the correct pattern here is to have the old tagList pending as delete and the new one pending as add
```suggestion
                    [newName]: {...oldPolicyTags, name: newName, pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD},
                    [oldName]: 
 {pendingAction:CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE},
```

Comment (luacmartins) [reply src/libs/actions/Policy.ts:2621]: In addition, we'd have successData to remove the pendingActions and delete the old taglist

**Review by luacmartins — CHANGES_REQUESTED** [review]
Left a few comments

**Comment by waterim** [comment]
@luacmartins Was it some changes related to workspaces?
Because I can't reach tags/categories at all, was thinking my changes causing it, but tried on staging on a new account but still the same
image

**Comment by waterim** [comment]
Oh hold till I can test m changes

File: src/libs/actions/Policy.ts (hunk 2 above)
Comment (waterim) [reply src/libs/actions/Policy.ts:2601]: errors are working, it shows an error, we have the same in categories
Maybe Im missing something

**Comment by luacmartins** [comment]
@mkhutornyi are you available to continue your review here? You can hardcode `isPolicyExpenseChatEnabled = true` to be able to see the policies

**Review by luacmartins — CHANGES_REQUESTED** [review]
Another issue that I see is that renaming the tag list temporarily duplicates tags

https://github.com/Expensify/App/assets/22219519/b30aa02b-d172-4d98-9ca9-7b9eef284d7a

**Comment by waterim** [comment]
@luacmartins @mkhutornyi 
Fixed double data, previously we were saving both old and new tagList in onyx, but now only the new one is there, but if failure happen -> old taglist will be added and new one will be deleted.

**Comment by luacmartins** [comment]
@mkhutornyi are you able to review this again now?

**Comment by luacmartins** [comment]
>Also, EnablePolicyTags api failing and calls are repeated

I think that's unrelated to this PR. That API hasn't been deployed to staging/production yet so it's expected that it'll fail for now. We're ok with that given that only a small number of internal users have access to these pages now.

**Comment by luacmartins** [comment]
@mkhutornyi just want to make sure you're on the right branch, because the button is there for me
Screenshot 2024-03-11 at 12 02 41 PM

**Comment by waterim** [comment]
And for me works too: 
image

**Comment by waterim** [comment]
I have this check when no policyTags: 
image

**Comment by luacmartins** [comment]
@waterim let's address the padding on that button

**Comment by waterim** [comment]
@mkhutornyi Added, that was a new change on category, styling there is changing sometimes

**Comment by luacmartins** [comment]
Thanks for flagging that @mkhutornyi. I think we shouldn't block the PR on that and we can address it when we work on the next view for Tags.

File: src/libs/actions/Policy.ts (hunk 2 above)
Comment (luacmartins) [reply src/libs/actions/Policy.ts:2664]: I'm retesting this now

**Comment by waterim** [comment]
@mkhutornyi This one is already in development in moreFeatures PR

File: src/libs/actions/Policy.ts (hunk 2 above)
Comment (luacmartins) [reply src/libs/actions/Policy.ts:2664]: No, this is still happening. I think we can continue with this and I'll address it in a fast following PR.

**Comment by luacmartins** [comment]
@mkhutornyi I found that issue too and my PR will fix that

**Review by mkhutornyi — APPROVED** [review]
@waterim please note all the reported bugs in this PR and make sure to cover them in next PR

**Review by luacmartins — APPROVED** [review]

**Comment by luacmartins** [comment]
Tests were passing. Removing label.

**Comment by lanitochka17** [comment]
This PR is failing because of issue https://github.com/Expensify/App/issues/38188 

The issue is reproducible in: All platforms


https://github.com/Expensify/App/assets/78819774/1adf2cd2-b78b-4d70-8752-8c1271871acb

File: src/pages/workspace/tags/WorkspaceTagsPage.tsx [hunk 4]
```diff
@@ -106,11 +143,28 @@ function WorkspaceTagsPage({policyTags, route}: WorkspaceTagsPageProps) {
...
                     </View>
-                    {tagList.length ? (
+                    {isLoading && (
+                        <ActivityIndicator
+                            size={CONST.ACTIVITY_INDICATOR_SIZE.LARGE}
+                            style={[styles.flex1]}
+                            color={theme.spinner}
+                        />
+                    )}
+                    {tagList.length === 0 && !isLoading && (
```
Comment (allroundexperts) [inline src/pages/workspace/tags/WorkspaceTagsPage.tsx:160]: This line caused a [bug](https://github.com/Expensify/App/issues/42220#issuecomment-2124305407) later on. The deleted tags in offline mode had there state set as `pending`.

**Pull request merged**